src/common/snippets/src/lowered/pass/split_loops.cpp

```diff
@@ -37,50 +41,106 @@ bool SplitLoops::can_be_split(const UnifiedLoopInfoPtr& loop_to_split, const Uni
 
 bool SplitLoops::run(LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, lowered::LinearIR::constExprIt end) {
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::SplitLoops")
+    auto can_be_fused_after_splitting =
+        [](const UnifiedLoopInfoPtr& parent_loop, const UnifiedLoopInfoPtr& loop, bool split_parent) {
+            // Note: we make a copy of loop infos to imitate the successfull splitting
+            // and check if such loops could be fused
+            const auto upper_loop = std::make_shared<UnifiedLoopInfo>(*parent_loop);
+            const auto lower_loop = std::make_shared<UnifiedLoopInfo>(*loop);
+            if (split_parent) {
+                upper_loop->set_increment(loop->get_increment());
+            } else {
+                lower_loop->set_increment(parent_loop->get_increment());
+            }
+            return FuseLoops::can_be_fused(upper_loop, lower_loop);
+        };
+
     const auto& loop_manager = linear_ir.get_loop_manager();
+    auto mark_splittable_loop_pair = [&](size_t current_loop_id,
+                                         size_t parent_loop_id,
+                                         std::vector<std::pair<size_t, size_t>>& loops_to_split,
+                                         bool allow_loop_to_fuse_with_tail) -> bool {
+        const auto current_loop = loop_manager->get_loop_info<UnifiedLoopInfo>(current_loop_id);
+        const auto parent_loop = loop_manager->get_loop_info<UnifiedLoopInfo>(parent_loop_id);
+        const bool split_parent = parent_loop->get_increment() < current_loop->get_increment();
+
+        // We don't split loops which are not compatible with parent loop because such loops will not be fused
+        if (!can_be_fused_after_splitting(parent_loop, current_loop, split_parent)) {
+            return false;
+        }
+
+        const auto& loop_to_split = split_parent ? parent_loop : current_loop;
+        const auto& loop_to_fuse = !split_parent ? parent_loop : current_loop;
```

Comment:
```suggestion
    const auto& [loop_to_split, loop_to_fuse] =
        split_parent ? std::tie(parent_loop,  current_loop)
                     : std::tie(current_loop, parent_loop);
```
Could this be better for understanding? What do you think?

Reply:
This is definitely better, thanks for the proposal, implemented